### ISSUE TITLE
[18:24:04] [Server thread/CRITICAL]: ErrorException: "Invalid argument supplied for foreach()" (EXCEPTION) in "MyStats.phar/src/MyStats/Util/DataManager" at line 99

### ISSUE THREAD

**Issue by matatacmca** [CzechPMDevs/MyStats#3]
Hi There

I got the below error when installing the MyStats plugin from PHAR 

```
[18:24:04] [Server thread/INFO]: Loading pocketmine.yml...
[18:24:04] [Server thread/INFO]: Loading server properties...
[18:24:04] [Server thread/INFO]: Selected English (eng) as the base language
[18:24:04] [Server thread/INFO]: Starting Minecraft: PE server version v1.2.0.81
...
[18:24:04] [Server thread/INFO]: This server is running PocketMine-MP version 1.7dev-83 "[REDACTED]" (API 3.0.0-ALPHA8)
[18:24:04] [Server thread/INFO]: PocketMine-MP is distributed under the LGPL License
...
[18:24:04] [Server thread/INFO]: Loading MyStats v1.4.2
[18:24:04] [Server thread/INFO]: Loading EconomyAPI v5.7.1-2.alpha8
[18:24:04] [Server thread/INFO]: Loading EconomyJob v2.0.4-1.alpha8
[18:24:04] [Server thread/INFO]: Enabling MyStats v1.4.2
[18:24:04] [Server thread/INFO]: Disabling MyStats v1.4.2
[18:24:04] [Server thread/CRITICAL]: ErrorException: "Invalid argument supplied for foreach()" (EXCEPTION) in "MyStats.phar/src/MyStats/Util/DataManager" at line 99
[18:24:04] [Server thread/DEBUG]: #0 MyStats.phar/src/MyStats/Util/DataManager(99): pocketmine\{closure}(integer 2, string Invalid argument supplied for foreach(), string phar:///var/minecraft/PMMP/plugins/MyStats.phar/src/MyStats/Util/DataManager.php, integer 99, array Array())
[18:24:04] [Server thread/DEBUG]: #1 MyStats.phar/src/MyStats/MyStats(83): MyStats\Util\DataManager->saveData()
[18:24:04] [Server thread/DEBUG]: #2 src/pocketmine/plugin/PluginBase(92): MyStats\MyStats->onDisable()
[18:24:04] [Server thread/DEBUG]: #3 src/pocketmine/plugin/PharPluginLoader(137): pocketmine\plugin\PluginBase->setEnabled(boolean )
[18:24:04] [Server thread/DEBUG]: #4 src/pocketmine/plugin/PluginManager(687): pocketmine\plugin\PharPluginLoader->disablePlugin(MyStats\MyStats object)
[18:24:04] [Server thread/DEBUG]: #5 MyStats.phar/src/MyStats/MyStats(121): pocketmine\plugin\PluginManager->disablePlugin(MyStats\MyStats object)
[18:24:04] [Server thread/DEBUG]: #6 MyStats.phar/src/MyStats/MyStats(63): MyStats\MyStats->check()
[18:24:04] [Server thread/DEBUG]: #7 src/pocketmine/plugin/PluginBase(90): MyStats\MyStats->onEnable()
[18:24:04] [Server thread/DEBUG]: #8 src/pocketmine/plugin/PharPluginLoader(122): pocketmine\plugin\PluginBase->setEnabled(boolean 1)
[18:24:04] [Server thread/DEBUG]: #9 src/pocketmine/plugin/PluginManager(610): pocketmine\plugin\PharPluginLoader->enablePlugin(MyStats\MyStats object)
[18:24:04] [Server thread/DEBUG]: #10 src/pocketmine/Server(1957): pocketmine\plugin\PluginManager->enablePlugin(MyStats\MyStats object)
[18:24:04] [Server thread/DEBUG]: #11 src/pocketmine/Server(1943): pocketmine\Server->enablePlugin(MyStats\MyStats object)
[18:24:04] [Server thread/DEBUG]: #12 src/pocketmine/Server(1652): pocketmine\Server->enablePlugins(integer 0)
[18:24:04] [Server thread/DEBUG]: #13 src/pocketmine/PocketMine(553): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string phar:///var/minecraft/PMMP/PocketMine-MP.phar/, string /var/minecraft/PMMP/, string /var/minecraft/PMMP/plugins/)
[18:24:04] [Server thread/DEBUG]: #14 /var/minecraft/PMMP/PocketMine-MP.phar(1): require(string phar:///var/minecraft/PMMP/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
[18:24:04] [Server thread/CRITICAL]: [MyStats] Download plugin from github! (https://github.com/CzechPMDevs/MyStats/releases)
[18:24:04] [Server thread/INFO]: [MyStats] &5&l[ &r&2MyStats &l&5]Submit issue to https://github.com/CzechPMDevs/MyStats/issues to fix it.
```

thanks

**Comment by GamakCZ** [comment]
Did you downloaded version 1.4.2 from releases?

**Comment by matatacmca** [comment]
I Downloaded from https://poggit.pmmp.io/r/12635/MyStats.phar

**Comment by ghost** [comment]
Hi, try updating your pmmp version

**Comment by GamakCZ** [comment]
Fixed, try to download version 1.4.3

**Comment by MondTV** [comment]
Is this commitable with 1.4.0 Servers
